### PR TITLE
Aiodns support

### PR DESCRIPTION
**Pull request by lphuberdeau**
## What these changes does?

Allows to use an alternate method for DNS resolution. The default resolution remains to the getaddrinfo system call using thread processors.

Alternate technique is to use aiodns instead, which allows to specify the nameservers to use while resolving. aiodns allows for higher concurrency.

## Related issue number

* Resolves issue #728 
* Follow-up to PR #747

## Checklist

- [x] CI testing works under windows

### CODE REVIEW

**Comment by gwillem** [comment]
NB Another benefit of aiodns is that is allows higher concurrency when crawling the Internet. So 100-200 crawlers per core should be possible.

**Comment by lphuberdeau** [comment]
Seems like the issue with appveyor/c-ares is limited to py35-64.

File: aiohttp/abc.py [hunk 1]
```diff
@@ -55,3 +55,15 @@ def __iter__(self):
         @abstractmethod
         def __await__(self):
             return
+
+
+class AbstractResolver(metaclass=ABCMeta):
```
Comment (asvetlov) [inline aiohttp/abc.py]: Just inherit from `ABC` like other classes from the module do.

File: aiohttp/abc.py [hunk 2]
```diff
@@ -55,3 +55,15 @@ def __iter__(self):
...
+
+    @asyncio.coroutine
+    @abstractmethod
+    def resolve(self, hostname):
+        """Return IP address for given hostname"""
+
+    @abstractmethod
+    def close(self):
```
Comment (asvetlov) [inline aiohttp/abc.py:69]: Please make `close` a coroutine -- my guts feel we will need it sometimes.

**Comment by lphuberdeau** [comment]
@asvetlov Requested changes are applied

**Comment by asvetlov** [comment]
Looks good but please fix a merge conflict first.

File: tests/test_resolver.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    aiodns = None
+
+
+class FakeResult:
+    def __init__(self, host):
+        self.host = host
+
+
+@pytest.fixture
+def loop():
```
Comment (jettify) [inline tests/test_resolver.py]: I think `loop` fixture from `conftest.py` could be used here

**Comment by lphuberdeau** [comment]
@asvetlov Branch is now up to date, using the common loop fixture.

**Comment by asvetlov** [comment]
Thanks

**Pull request merged**